exasol/analytics/query_handler/query/select.py

```diff
@@ -15,6 +17,34 @@ def query_string(self) -> str:
         return self._query_string
 
 
+class WriteQuery(SelectQuery):
+    """
+    A query wrapper for a query that changes data in a database table (e.g. INSERT or
+    UPDATE) or creates the table (e.g. CREATE TABLE). This type of query is auditable.
+    """
+    def __init__(self, query_string: str, affected_table: str, audit: bool = False):
+        super().__init__(query_string)
+        self._affected_table = affected_table
+        self._audit = audit
+
+    @property
+    def affected_table(self) -> str:
+        return self._affected_table
+
+    @property
+    def audit(self) -> bool:
+        return self._audit
+
+
+class AuditQuery(SelectQuery):
+    """
+    A query wrapper for a special read-only query that selects data for auditing.
+    The output columns of such a query should correspond to the definition of the
+    audit data provided by a QueryHandler.
```

Comment:
I am not sure about, who actually defines the resp. structure. 
Is it an arbitrary query handler?
Can different query handlers have different audit columns?

Reply:
It's the QueryHandlerAudit

Reply:
In my opinion, yes, different QueryHandlers can have different audit columns. That's why I wrote, QueryHandlers should have a function to tell which additional audit columns it would provide. This way we can go the call stack down and collect all columns. However, there should be a fixed subset, that all provide. The main difficulty is how to collect the value over the call stack, because I would like to have, that the leaves don't need to know anything about the columns provided by the QueryHandler in the middle.  Current, idea is a kind of Matroska QueryHandlerAudit, where each step wrap the QueryHandlerAudit it received from its child into its own.

Reply:
This is something more complicated than what I had in mind. My idea was that the audit fields provided by a Query Handler (all or most are optional) are defined in the QueryHandlerAudit dataclass. Then there are two ways for the QH to pass them to the Executor. 1. As an instance of this class. 2. As a query that once executed will return columns corresponding to the fields of the dataclass.

Reply:
So while you are executing queries in the `query_list` you can see a query wrapped in the `AuditQuery` wrapper. Then you transform this query into f'INSERT INTO AuditTable {the_query}'